packages/cdk/resources/Cognito.ts

```diff
@@ -0,0 +1,207 @@
+  readonly mockOidcUserInfoEndpoint?: string
+  readonly mockOidcjwksEndpoint?: string
+  readonly mockTokenEndpoint: string
+  readonly shortCognitoDomain: string
+  readonly fullCloudfrontDomain: string
+  readonly fullCognitoDomain: string
+  readonly cognitoCertificate: ICertificate
+  readonly hostedZone: IHostedZone
+}
+
+/**
+ * AWS Cognito User Pool and supporting resources
+ */
+export class Cognito extends Construct {
+  public readonly userPool: UserPool
+  public readonly userPoolClient: UserPoolClient
+  public readonly userPoolDomain: UserPoolDomain
+  public readonly primaryPoolIdentityProvider: UserPoolIdentityProviderOidc
+  public readonly mockPoolIdentityProvider: UserPoolIdentityProviderOidc
+
+  public constructor(scope: Construct, id: string, props: CognitoProps) {
+    super(scope, id)
+
+    // Resources
+    const userPool = new UserPool(this, "UserPool", {
+      removalPolicy: RemovalPolicy.DESTROY
+    })
+
+    const userPoolDomain = new UserPoolDomain(this, "UserPoolDomain", {
+      userPool,
+      customDomain: {
+        domainName: props.fullCognitoDomain,
+        certificate: props.cognitoCertificate
+      }
+    })
+
+    new ARecord(this, "UserPoolCloudFrontAliasIpv4Record", {
+      zone: props.hostedZone,
+      recordName: props.shortCognitoDomain,
+      target: RecordTarget.fromAlias(new UserPoolDomainTarget(userPoolDomain))
+    })
+
+    new AaaaRecord(this, "UserPoolCloudFrontAliasIpv6Record", {
+      zone: props.hostedZone,
+      recordName: props.shortCognitoDomain,
+      target: RecordTarget.fromAlias(new UserPoolDomainTarget(userPoolDomain))
+    })
+
+    // these are the endpoints that are added to user pool identity provider
+    // note we override the token endpoint to point back to our custom token
+    const oidcEndpoints: OidcEndpoints = {
+      authorization: props.primaryOidcAuthorizeEndpoint,
+      jwksUri: props.primaryOidcjwksEndpoint,
+      token: `https://${props.fullCloudfrontDomain}/api/token`,
+      userInfo: props.primaryOidcUserInfoEndpoint
+    }
+
+    // eslint-disable-next-line max-len
+    // see https://digital.nhs.uk/services/care-identity-service/applications-and-services/cis2-authentication/guidance-for-developers/detailed-guidance/scopes-and-claims
+    // about claims that we need
+    const primaryPoolIdentityProvider = new UserPoolIdentityProviderOidc(this, "UserPoolIdentityProvider", {
+      name: "Primary", // this name is used in the web client
+      clientId: props.primaryOidcClientId,
+      clientSecret: props.primaryOidClientSecret,
+      issuerUrl: props.primaryOidcIssuer,
+      userPool: userPool,
+      attributeRequestMethod: OidcAttributeRequestMethod.GET,
+      scopes: ["openid", "profile", "email", "nhsperson", "nationalrbacaccess"],
+      endpoints: oidcEndpoints
+    })
+
+    const supportedIdentityProviders: Array<UserPoolClientIdentityProvider> = [
+      UserPoolClientIdentityProvider.COGNITO,
```

Comment:
Why are we allowing Cognito itself to be an identity provider? I thought we said that real CIS2 and mock auth would be all we need.

Reply:
removed